restli-common/src/main/java/com/linkedin/restli/internal/common/URIParamUtils.java

```diff
@@ -486,6 +512,7 @@ public static URI replaceQueryParam(URI uri, String queryParam, DataComplex valu
    */
   public static void addSortedParams(UriBuilder uriBuilder, DataMap params)
   {
+    convertRawProjectionFieldsToMaskTree(params);
```

Comment:
Agree with Evan, also the projection parameters are already available as MaskTree in resourContext. So a better fix would be to update the `buildCriteriaURI()` method in BatchFinderResponseBuilder to replace the projection params with masktree objects.

Reply:
Yes, changed the way I extract projection MaskTree by directly referencing those projection fields which already in ResourceContextImpl